daemon/cmd/daemon_main.go

```diff
@@ -360,6 +360,15 @@ func initializeFlags() {
 	flags.String(option.NodeEncryptionOptOutLabels, defaults.NodeEncryptionOptOutLabels, "Label selector for nodes which will opt-out of node-to-node encryption")
 	option.BindEnv(Vp, option.NodeEncryptionOptOutLabels)
 
+	flags.Bool(option.EnableEncryptionStrictMode, false, "Enable encryption strict mode")
+	option.BindEnv(Vp, option.EnableEncryptionStrictMode)
+
+	flags.String(option.EncryptionStrictModeCIDR, "", "In strict-mode encryption, all unencrypted traffic coming from this CIDR and going to this same CIDR will be dropped")
+	option.BindEnv(Vp, option.EncryptionStrictModeCIDR)
+
+	flags.Bool(option.EncryptionStrictModeAllowRemoteNodeIdentities, false, "EncryptionStrictModeAllowRemoteNodeIdentities allows dynamic lookup of remote node identities. This is required when tunneling is used or direct routing is used and the node CIDR and pod CIDR overlap.")
```

Comment:
nit: The description here is a bit unwieldy. The use of the internal CamelCase name also something I'd recommend not to do (from a user's perspective, it's a [kebab-case](https://it.wikipedia.org/wiki/Kebab_case) name).

```suggestion
	flags.Bool(option.EncryptionStrictModeAllowRemoteNodeIdentities, false, Allows unencrypted traffic from pods to remote node identities within the strict mode CIDR. This is required when tunneling is used or direct routing is used and the node CIDR and pod CIDR overlap.")
```

While I understand the use of the encryption prefix, the flag is now very long. Maybe `encryption-strict-mode-allow-remote-nodes`? Even then they are still a bit long.

Reply:
Thanks for the improvement! I also see that the names are quite long now. But I think it is since their naming is quite descriptive and the flag itself is hidden from the user anyway, since they configure it through the helm values, right? Therefore, I'd leave the name as is in favor of the descriptiveness.